From dfc0fd2eec82d5363f0a331cd71a52a5ec682ece Mon Sep 17 00:00:00 2001
From: Lulu Cheng <chenglulu@loongson.cn>
Date: Thu, 24 Jul 2025 19:07:25 +0800
Subject: [PATCH 63/89] LoongArch: Remove the definition of
 CASE_VECTOR_SHORTEN_MODE.

On LoongArch, the switch jump-table always stores absolute
addresses, so there is no need to define the macro
CASE_VECTOR_SHORTEN_MODE.

gcc/ChangeLog:

	* config/loongarch/loongarch.h
	(CASE_VECTOR_SHORTEN_MODE): Delete.
---
 gcc/config/loongarch/loongarch.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/gcc/config/loongarch/loongarch.h b/gcc/config/loongarch/loongarch.h
index 10b5a0b0f32..62307b94b53 100644
--- a/src/gcc/config/loongarch/loongarch.h
+++ b/src/gcc/config/loongarch/loongarch.h
@@ -824,8 +824,6 @@ typedef struct {
 
 #define CASE_VECTOR_MODE Pmode
 
-#define CASE_VECTOR_SHORTEN_MODE(MIN, MAX, BODY) Pmode
-
 /* Define this as 1 if `char' should by default be signed; else as 0.  */
 #ifndef DEFAULT_SIGNED_CHAR
 #define DEFAULT_SIGNED_CHAR 1
-- 
2.33.0

